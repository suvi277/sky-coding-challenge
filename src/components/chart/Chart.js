import React from 'react';
import { Line as LineChart } from 'react-chartjs-2';
import { dataSetsConfig, options } from './chartConfig';
import Aux from 'components/hoc/Aux';

const Chart = ({ labels, lineData, day }) => {
	const chartData = {
		labels: labels,
		datasets: [
			{
				...dataSetsConfig,
				data: lineData
			}
		]
	};

	return (
		<Aux>
			<h4 className="text-center">{day}</h4>
			<LineChart data={chartData} options={options} width="600" height="250" />
		</Aux>
	);
};
export default Chart;
