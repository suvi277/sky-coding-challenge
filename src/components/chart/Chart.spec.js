import React from 'react';
import { shallow } from 'enzyme';
import Chart from './Chart';
import { Line as LineChart } from 'react-chartjs-2';

describe('Navigation Component', () => {
	const props = {
		labels: [],
		lineData: [],
		day: ''
	};

	const wrapper = shallow(<Chart {...props} />);

	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should exist Navigation Component', () => {
		expect(wrapper.exists(LineChart)).toBeTruthy();
	});
});
