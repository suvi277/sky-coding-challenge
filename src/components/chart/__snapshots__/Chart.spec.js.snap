// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Component renders as expected 1`] = `
<Aux>
  <h4
    className="text-center"
  />
  <Line
    data={
      Object {
        "datasets": Array [
          Object {
            "borderColor": "#FF550B",
            "borderWidth": 3,
            "data": Array [],
            "fill": false,
            "label": "Data",
            "pointBackgroundColor": "#FF550B",
            "pointBorderWidth": 3,
            "pointHoverBackgroundColor": "#FFFFFF",
          },
        ],
        "labels": Array [],
      }
    }
    height={250}
    options={
      Object {
        "legend": Object {
          "display": false,
        },
        "scales": Object {
          "xAxes": Array [
            Object {
              "gridLines": Object {
                "borderDash": Array [
                  2,
                  5,
                ],
                "lineWidth": 2,
                "zeroLineColor": "#EEEEEE",
                "zeroLineWidth": 1,
              },
              "scaleLabel": Object {
                "display": true,
                "fontColor": "#563D7C",
                "fontSize": 25,
                "labelString": "Values",
                "lineHeight": 3,
              },
              "ticks": Object {
                "beginAtZero": true,
                "fontColor": "#666666",
                "fontSize": 14,
              },
            },
          ],
          "yAxes": Array [
            Object {
              "gridLines": Object {
                "borderDash": Array [
                  2,
                  5,
                ],
                "lineWidth": 2,
                "zeroLineColor": "#EEEEEE",
                "zeroLineWidth": 1,
              },
              "scaleLabel": Object {
                "display": true,
                "fontColor": "#563D7C",
                "fontSize": 25,
                "labelString": "Values",
                "lineHeight": 3,
              },
              "ticks": Object {
                "beginAtZero": true,
                "fontColor": "#666666",
                "fontSize": 14,
              },
            },
          ],
        },
        "tooltips": Object {
          "backgroundColor": "rgba(0,122,92, 0.8)",
          "displayColors": false,
          "yAlign": "bottom",
        },
      }
    }
    width={600}
  />
</Aux>
`;
